### PR TITLE
Cypress code coverage

### PR DESCRIPTION
**Pull request by Jin1L**
This PR contains a set of e2e testings that were written with cypress which focuses on UI components and FE usage.

For creating a test, please check CYPRESS.md file

### Preset & Configuration **

The most troublesome part of the test development was setting up exportation of coverage results. In case anything goes wrong with configuring current test environment in the future, here are the steps for configuration:
1. make sure `@cypress/code-coverage` is installed and confirm:
 - `@cypress/code-coverage/support` is imported in `cypress/support/e2e.ts`
 - `@cypress/code-coverage/task` is required in `cypress/plugins/index.ts`
 2. Since studio uses`vite` as the build tool and istanbuljs is a test coverage tool that works well with cypress, confirm:
 -  `vite-plugin-istanbul` is installed
 - `istanbul` is imported from `vite-plugin-istanbul` into `vite.config.js`
 -  `istanbul(["src/**/*.ts", "src/**/*.tsx"])` is included as part of plugins nested inside `defineConfig()` in `vite.config.js`.
 3. make sure `@istanbuljs/nyc-config-typescript` is installed and confirm:
 - this dependency is extended by `nyc` (a package that produces test coverage stats) in `package.json`
 4. Finally, in `cypress.config.ts` ensure that `./cypress/plugins/index.ts` is required inside `setupNodeEvents()` function.
 
### Testing  

In order to test UI components (ex: Sidebar, NodeModal) from cypress, the data test attributes (i.e unique identifiers) must be added to the corresponding `.tsx` files.

Example: `data-testid="component-name"` or `data-cy="component-name"`.

For visual tests, before writing any test lines, the setup below has to be made. Snapshots that are successfully captured will be exported to Applitools and it will conduct additional tests.
```ts
  // This method performs setup before each test.
  beforeEach(() => {
    // Open Eyes to start visual testing.
    cy.eyesOpen({
      // Set this name wisely: Applitools features rely on a shared app name across tests.
      appName: "studio",
      testName: Cypress.currentTest.title
    });
  });
```

### CI/CD

Throughout commits,  some python tests for nodes folder unreasonably failed on github action. If this occurs when making changes, add `--ignore=nodes` flag beside `python-tests | run |` command in `ci.yml`.

### CODE REVIEW

**Comment by yassine-moumen** [comment]
## To access Flojoy-studio build from this pull request, you can use the following URLs: 

Windows installation: http://623_windows.flojoy-test.com:3000 
Linux installation: http://623_linux.flojoy-test.com:3000 

Please note that it may take a few minutes for the instance to start up, so please be patient. 

It is important to know that the instance will automatically shut down after 4 hours to conserve resources. If you wish to start the instance again after it shuts down (or restart it to pickup new changes), you can use this API : 

Windows: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-0a0a9481ab469d5c3&domain=623_windows 

Linux: https://kgpsoscpc4q52d4mtwuu7jhvie0qdiso.lambda-url.us-east-1.on.aws/start_instance?instance_id=i-04e1ee0e820d5361f&domain=623_linux

File: package.json [hunk 1]
```diff
@@ -175,14 +175,38 @@
...
     "exclude": [
-      "assets/",
-      "test/"
+      "src/__tests__",
+      "src/hooks/__mocks__",
+      "src/utils/__mocks__",
+      "src/utils/CustomError.ts",
+      " src/feature/controls_panel/",
+      " src/feature/results_panel/",
+      "src/feature/flow_chart_panel/components/nodes/PieChart.tsx",
```
Comment (scorinaldi) [inline package.json:192]: Ideally we want this exclude list to be small. Please go grep through this list and confirm that none of the files/classes here are being used in prod. If there's merged (non-test) code that's not being used in in prod that's not great either, so we should file an issue to follow up later.

If you wind up un-commenting some stuff and the coverage falls below 80% that's ok. We just need to make a follow up list so we know what tests to write in the future.

File: src/__tests__/feature/flow_chart_panel/__snapshots__/FlowChartTabView.test.tsx.snap [hunk 2]
```diff
@@ -260,25 +263,7 @@ exports[`FlowChartTabView should render the component correcty 1`] = `
                 Export Result
               </button>
               <button
-                style="display: flex; gap: 10.77px;"
```
Comment (scorinaldi) [inline src/__tests__/feature/flow_chart_panel/__snapshots__/FlowChartTabView.test.tsx.snap:263]: These are auto-generated , we don't know why it was removed.

**Review by scorinaldi — CHANGES_REQUESTED** [review]
This mostly looks good, thanks for your hard work here! There are a few remaining issues, which I'll list below:

- Please write a thorough Top Comment for this PR. If you had to review this change in a year, what would you present to yourself to make it easy as possible to have them understand the change? to see an outline of the solution in the PR top comment. How did you solve this problem? What are notable pieces of code, etc. 
- Please write an intro guide in Notion to help folks understand how to write FE tests. You spent the time and effort to learn this from scratch, your mission is to make sure **no one ever has to do that again**. We want everyone to write front FE tests!!
- Please add a 1-2 sentence description in each .cy.ts test file to say what the test does
- Please run lint on this.

(Note to other reviewers that the tsx changes are just 1 liners, just identifying the components to the testing framework)

**Review by scorinaldi — APPROVED** [review]
Comments and top comment look great! Thanks for all your hard work here.

LGTM

**Pull request merged**